components/ntp_background_images/browser/new_tab_takeover_infobar_util_unittest.cc

```diff
@@ -0,0 +1,118 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/ntp_background_images/browser/new_tab_takeover_infobar_util.h"
+
+#include "base/test/scoped_feature_list.h"
+#include "brave/components/brave_ads/core/public/ad_units/new_tab_page_ad/new_tab_page_ad_feature.h"
+#include "brave/components/brave_rewards/core/pref_names.h"
+#include "brave/components/brave_rewards/core/pref_registry.h"
+#include "brave/components/ntp_background_images/common/pref_constants.h"
+#include "brave/components/ntp_background_images/common/pref_names.h"
+#include "brave/components/ntp_background_images/common/view_counter_pref_registry.h"
+#include "components/sync_preferences/testing_pref_service_syncable.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace ntp_background_images {
+
+class NewTabTakeoverInfobarUtilTest : public testing::Test {
+ public:
+  NewTabTakeoverInfobarUtilTest() {
+    RegisterProfilePrefs(pref_service_.registry());
+    brave_rewards::RegisterProfilePrefs(pref_service_.registry());
+  }
+
+  ~NewTabTakeoverInfobarUtilTest() override = default;
+
+  PrefService* pref_service() { return &pref_service_; }
+
+  void EnableSupportNewTabPageAdConfirmationsForNonRewards() {
+    scoped_feature_list_.InitWithFeaturesAndParameters(
+        {{brave_ads::kNewTabPageAdFeature,
+          {{"should_support_confirmations_for_non_rewards", "true"}}}},
+        {});
+  }
+
+  void DisableSupportNewTabPageAdConfirmationsForNonRewards() {
+    scoped_feature_list_.InitWithFeaturesAndParameters(
+        {{brave_ads::kNewTabPageAdFeature,
+          {{"should_support_confirmations_for_non_rewards", "false"}}}},
+        {});
+  }
```

Comment:
Could we maybe have a single function where we pass true/false, something like `EnableShouldSupportNewTabPageAdConfirmationsForNonRewardsFeature(...)`? These functions are also used in another test, maybe we could add a test util?

Reply:
Fixed and filed a follow-up issue: https://github.com/brave/brave-browser/issues/45934